### DISCUSSION TITLE
Multi-Screen Capture

### DISCUSSION THREAD

**Discussion by ollieanwyll** [mediar-ai/screenpipe#190] [Ideas]
I've purchased the MacOS App and was a little disappointed that it doesn't capture all of my screens.
Also, to get screen-pipe to work, I had to disable my iPad sidecar as an extra monitor.
I have 4 screens in total and am using a MBP M3 Pro.

Can we get support for this and provide a workaround for still using iPad Sidecar?
I'm the first person in my team of 140 people to test this, and we all have multiple monitors.
Multi-Screen recording will be a must if we were to all purchase the MacOS App.

**Comment by louis030195** [comment]
will do

**Comment by m13v** [comment]
I added a loop to go through all monitors and capture screenshots of all visible windows, works on my end, pushing to the main now

**Comment by louis030195** [comment]
@ollieanwyll also curious if recording from a server would be a fit for you?

e.g. your local machine forward your monitors and audio devices to your own server or ours that runs screenpipe and do the processing there?

**Comment by ollieanwyll** [comment]
Hey Louis,

Sounds great.

I can see how my machine handles it; I could configure a server if it's
struggling.


Are you thinking of just a regular CPU-based Linux cloud instance?

Many thanks,
Ollie

**Comment by m13v** [comment]
Hey @ollieanwyll, please check if the screen captured started working for your other monitors. We still make recording for one screen but we capture all visible windows from all monitors now

**Comment by louis030195** [comment]
@ollieanwyll would love to hear any news since we have well supported multi monitor
